### PR TITLE
Fix dead nested assignment detected by clang CI

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by wargio — APPROVED** [review]

File: librz/hash/sha2.c [hunk 1]
```diff
@@ -455,9 +455,6 @@ void SHA256_Transform(RZ_SHA256_CTX *context, const ut32 *data) {
 	context->state[5] += f;
 	context->state[6] += g;
 	context->state[7] += h;
-
-	/* Clean up */
```
Comment (ret2libc) [inline librz/hash/sha2.c:459]: At line 380 I see similar pattern. Is that dead as well?

File: librz/util/print.c [hunk 2]
```diff
@@ -1194,13 +1194,11 @@ RZ_API void rz_print_hexdump(RzPrint *p, ut64 addr, const ut8 *buf, int len, int
 					if (comment) {
 						if (p && p->colorfor) {
 							a = p->colorfor(p->user, addr + j, true);
-							if (a && *a) {
-								b = Color_RESET;
-							} else {
-								a = b = "";
+							if (!a || !*a) {
```
Comment (ret2libc) [inline librz/util/print.c:1197]: are you sure `b` is dead here? This is inside a loop and `b` is used up in the code.

Comment (ghost) [reply librz/util/print.c:1197]: b is used line 1026, 1028, 1030

defined line 1003, 1005, 1008

So everytime b is used it has been defined before (1003, 1005 or 1008)

File: librz/hash/sha2.c (hunk 1 above)
Comment (ghost) [reply librz/hash/sha2.c:459]: i removed all useless clean up comment

**Comment by ghost** [comment]
Rebase if possible

File: librz/util/print.c (hunk 2 above)
Comment (ret2libc) [reply librz/util/print.c:1197]: Good point, thanks for double checking ;)

**Review by ret2libc — APPROVED** [review]

**Pull request merged**